CLAUDE.md

```diff
@@ -0,0 +1,276 @@
+variants in downstream analysis.
+
+# High-Level Architecture (WDLs, BigQuery, Terra)
+
+## Java Artifacts
+
+GVS is written in a mixture of Java, Python, WDL (Workflow Description
+Language), and shell scripts. The Java code can be built from the repository
+root using the `./gradlew clean shadowJar` command. This compilation requires a
+JDK of at least version 17. For local compilation, if
+[SDKMAN!](https://sdkman.io/) is available, one of the Java 17 versions can be
+used (most GATK builds seem to use Temurin).
+
+## Docker Versioning
+
+The `GvsUtils.wdl` workflow's `GetToolVersions` task is used to retrieve the
+version of various Docker images used by GVS. This task is usually run at the
+start of a GVS workflow to ensure that the correct versions of tools are being
+used.
+
+## Reference Versioning
+
+The `GvsUtils.wdl` workflow's `GetReference` task is used to retrieve the
+canonical paths to references used by GVS. This task is usually run at the start
+of a GVS workflow to ensure that the correct references are being used.
+
+## Docker artifacts
+
+GVS uses multiple Docker images to run its workflows. Some of these are specific
+to GVS as documented below.
+
+### GATK Docker Image
+
+This builds a GATK Docker image based on the GVS branch of the GATK codebase.
+Instructions for building this image can be found in the
+`scripts/variantstore/docs/Build Docker from VM/building_gatk_docker_on_a_vm.md`
+file. An x86 cloud-based VM is used to build this x86 Docker image, which is
+then pushed to the Broad's GAR (Google Artifact Registry) repository.
+
+### Variants Docker Image
+
+This is a Google Cloud SDK Alpine-based Docker image to which many useful tools,
+Python packages, and scripts have been added. This can be built on a developer's
+local machine using the `scripts/variantstore/scripts/build_docker.sh` script.
+
+### Nirvana Docker Image
+
+This is an Illumina Nirvana-based Docker image build from
+`scripts/variantstore/variant-annotations-table/build_docker.sh`. Since Nirvana
+is no longer being updated this image hopefully will also not need to be updated
+in the future.
+
+## "location" versus "position"
+
+GVS leverages BigQuery partitioning and clustering to optimize query performance
+and cost. The core tables are either partitioned by sample ID or location,
+depending on the access pattern for which they are optimized. GVS uses the term
+"location" in a very specific way: the chromosome number is multiplied by
+1,000,000,000,000 and then the position on the chromosome is added. For example,
+chromosome 1 position 1000 becomes `1000000000000 + 1000 = 1000000001000`. GVS
+locations use a chromosome number of 23 for the X chromosome and 24 for the Y
+chromosome. This location encoding allows for efficient storage and querying in
+BigQuery using a single partitionable value. GVS references to "position"
+usually refer to position on a given chromosome and do not mean the same thing
+as "location".
```

Comment:
My only suggestion is to also point it directly to SchemaUtils so it can see the implementation of location encoding directly.  We want it to use the existing implementation to encode and extract information into locations whenever possible, after all, and not have it writing its own versions on the fly according to the described formula. So add in a ref to SchemaUtils.encodeLocation and SchemaUtils.decodeChromosomeIndex and SchemaUtils.decodePosition